oidc-controller/api/core/acapy/tests/test_client.py

```diff
@@ -0,0 +1,216 @@
+import json
+
+import mock
+import pytest
+from api.core.acapy.client import (CREATE_PRESENTATION_REQUEST_URL,
+                                   PRESENT_PROOF_RECORDS,
+                                   PUBLIC_WALLET_DID_URI, WALLET_DID_URI,
+                                   AcapyClient)
+from api.core.acapy.config import MultiTenantAcapy, SingleTenantAcapy
+from api.core.acapy.models import CreatePresentationResponse, WalletDid
+from api.core.acapy.tests.__mocks__ import (create_presentation_response_http,
+                                            presentation_request_configuration)
+from api.core.config import settings
+
+
+@pytest.mark.asyncio
+@mock.patch.object(settings, "ACAPY_TENANCY", None)
+async def test_init_no_setting_returns_client_with_single_tenancy_config():
+    client = AcapyClient()
+    assert client is not None
+    assert isinstance(client.agent_config, SingleTenantAcapy) is True
+
+
+@pytest.mark.asyncio
+@mock.patch.object(settings, "ACAPY_TENANCY", "single")
+async def test_init_single_returns_client_with_single_tenancy_config():
+    client = AcapyClient()
+    assert client is not None
+    assert isinstance(client.agent_config, SingleTenantAcapy) is True
+
+
+@pytest.mark.asyncio
+@mock.patch.object(settings, "ACAPY_TENANCY", "multi")
+async def test_init_multi_returns_client_with_multi_tenancy_config():
+    client = AcapyClient()
+    assert client is not None
+    assert isinstance(client.agent_config, MultiTenantAcapy) is True
+
+
+@pytest.mark.asyncio
+async def test_create_presentation_returns_sucessfully_with_valid_data(requests_mock):
+    requests_mock.post(
+        settings.ACAPY_ADMIN_URL + CREATE_PRESENTATION_REQUEST_URL,
+        headers={},
+        json=json.dumps(create_presentation_response_http),
+        status_code=200,
+    )
+
+    with mock.patch.object(CreatePresentationResponse, "parse_obj", return_value={'result': 'success'}):
+        client = AcapyClient()
+        client.agent_config.get_headers = mock.MagicMock(
+            return_value={'x-api-key': ''})
+        presentation_request = client.create_presentation_request(
+            presentation_request_configuration)
+        assert presentation_request is not None
```

Comment:
Would it make sense to check some basic fields in the resulting presentation request, to ensure it matches expectations?

Reply:
I think that would be a good idea, yes. More info in my other comment. Currently is just checking for 200 response and response isn't empty. Other exceptions are unhandled.